doc/index.rst

```diff
@@ -9,100 +7,155 @@
 Matplotlib |release| documentation
 ##################################
 
```

Comment:
The changes to this page mean I need to scroll down two screens on my pretty large monitor to see any useful information.  I preferred the old version - this is drifting back to the old version with too much info on the front page.  The current version also only came in 3.5, I think it's not nice to change the landing page too often.